phantazm-commons/src/main/java/com/github/phantazmnetwork/commons/FileUtils.java

```diff
@@ -0,0 +1,102 @@
+package com.github.phantazmnetwork.commons;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Objects;
+import java.util.function.BiPredicate;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+import java.util.function.Consumer;
+
+/**
+ * Contains utility methods related to file IO.
+ */
+public final class FileUtils {
+    private FileUtils() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * A {@link Consumer}-like interface that can throw an {@link IOException}.
+     * @param <TType> the type of object to accept
+     */
+    public interface IOConsumer<TType> {
```

Comment:
can't believe i forgot to add @FunctionalInterface

Reply:
(and I just notified someone)

Reply:
I'd imagine @FunctionalInterface @NotNull @Nullable and @Override get notified all the time